src/type.cpp

```diff
@@ -83,3 +83,41 @@ std::unique_ptr<Type> ArrType::Clone() const {
 std::size_t ArrType::len() const {
   return len_;
 }
+
+bool FuncType::IsEqual(const Type& that) const noexcept {
+  if (const auto* that_func = dynamic_cast<const FuncType*>(&that)) {
+    if (that_func->param_types_.size() != param_types_.size()) {
+      return false;
+    }
+    if (!that_func->return_type_->IsEqual(*return_type_)) {
+      return false;
+    }
+    for (auto i = std::size_t{0}, e = param_types_.size(); i < e; ++i) {
+      if (!that_func->param_types_.at(i)->IsEqual(*param_types_.at(i))) {
+        return false;
+      }
+    }
+    return true;
+  }
+  return false;
+}
+
+std::size_t FuncType::size() const {
+  const auto pointer_size = 8;
+  return pointer_size;
+}
+
+std::string FuncType::ToString() const {
+  // FIXME: This is to not break the tests; should include the parameter types.
```

Comment:
Currently, our _ast dump_ only displays the _return type_ of the function, but it would be more comprehensive to include the _parameter types_ as well.

Consider the following program as an example:
```c
int main(int argc, char** argv) {
  return 0;
}
```

LLVM generates the following dump:
```
...
`-FunctionDecl 0x11e3340 <test.c:2:1, line:4:1> line:2:5 main 'int (int, char **)'
  |-ParmVarDecl 0x11e31b8 <col:10, col:14> col:14 argc 'int'
  |-ParmVarDecl 0x11e3260 <col:20, col:27> col:27 argv 'char **'
  `-CompoundStmt 0x11e3468 <col:33, line:4:1>
    `-ReturnStmt 0x11e3458 <line:3:3, col:10>
      `-IntegerLiteral 0x11e3438 <col:10> 'int' 0
...
```

Reply:
Noted!